File: wwwroot/js/modifyStructures.js
function EditStructures() {
    let properties = {
        Id: $(".modify-structure-Id").val(),
        InstituitionId: $(".evaluator-instituition").val(),
        DivisionId: $(".evaluator-division").val(),
        SectionId: $(".evaluator-section").val(),
        SectorId: $(".evaluator-sector").val(),
    };
    $.post("/Structure/ModifyStructures", properties)

        .done(function (output) {
            if (output.stats == 1) {
                $(location).attr('href', '../Edit/' + properties.Id);
            } else if (output.stats == 2) {

                if (properties.InstituitionId == "") {
                    $(".instituition-validation-error").html('Selecione um órgão!');
                } else {
                    $(".instituition-validation-error").html('');
                }
                if (properties.DivisionId == "") {
                    $(".division-validation-error").html('Selecione uma divisão!');
                } else {
                    $(".division-validation-error").html('');
                }
                if (properties.SectionId == "") {
                    $(".section-validation-error").html('Selecione uma seção!');
                } else {
                    $(".section-validation-error").html('');
                }
                if (properties.SectorId == "") {
                    $(".sector-validation-error").html('Selecione um setor!');
                } else {
                    $(".sector-validation-error").html('');
                }

            } else if (output.stats == 3) {
                return false;
            }
        })

        .fail(function () {
            alert("Ocorreu um erro!");
        });
}

$(document).ready(function () {
    $("#edit-structures-form").submit(function (e) {
        e.preventDefault();
        EditStructures();
    });
});

File: Models/StructuresModels/ModifyStructures.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SEP_Web.Models.StructuresModels;

public class ModifyStructures
{
    [Key, Required(ErrorMessage = "O campo id é obrigatório!")]
    public int Id { get; set; }

    [ForeignKey("InstituitionId"), Required(ErrorMessage = "Selecione um órgão!")]
    public int? InstituitionId { get; set; }
    public Instituition Instituition { get; set; }

    [ForeignKey("DivisionId"), Required(ErrorMessage = "Selecione uma divisão!")]
    public int? DivisionId { get; set; }
    public Division Division { get; set; }

    [ForeignKey("SectionId"), Required(ErrorMessage = "Selecione uma seção!")]
    public int? SectionId { get; set; }
    public Section Section { get; set; }

    [ForeignKey("SectorId"), Required(ErrorMessage = "Selecione um setor!")]
    public int? SectorId { get; set; }
    public Sector Sector { get; set; }
}